src/main/java/com/broadleafcommerce/autoconfigure/HSQLDBServer.java

```diff
@@ -64,20 +68,62 @@ public HSQLDBServer(final HSQLDBProperties autoProps, Environment environment) {
 
     @Override
     public boolean isRunning() {
-        try (Socket ignored = new Socket(InetAddress.getByName(null), props.getIntegerProperty("server.port", 0))) {
-            return true;
-        } catch (IOException ignored) {
-            return false;
+        boolean isRunning = false;
+        Connection c = null;
+
+        final int port = props.getIntegerProperty("server.port", 0);
+        try {
+            Class.forName("org.hsqldb.jdbc.JDBCDriver");
+            final String url = "jdbc:hsqldb:hsql://" + InetAddress.getByName(null).getHostName() + ":"
+                               + port + "/"
+                               + props.getProperty("server.dbname.0", "");
+            final String username = "SA";
+            final String password = "";
+            
+            c = DriverManager.getConnection(url, username, password);
+            
+            isRunning = true;
+        } catch (ClassNotFoundException e) {
+            LOG.error("ERROR: Cannot test if an HSQL server is running: failed to load HSQLDB JDBC driver.", e);
+        } catch (SQLException e) {
```

Comment:
Do we get the same exception if something is there and not responding correctly vs something isn't there at all?